### PR TITLE
Status module

### PR DESCRIPTION
**Pull request by DanielmTheDev**
I am open for any suggestions regarding those 3 functions in the module. Here a little demo:

Animation

### CODE REVIEW

File: src/spectrecoff-cli/Program.fs [hunk 1]
```diff
@@ -261,6 +261,21 @@ let main argv =
                     |> ignore
             ) |> ignore
 
+        config
+            .AddBranch("status", fun(branchConfig: IConfigurator<StatusSettings>) ->
+                branchConfig
+                    .AddCommand<StatusExample>("example")
+                    .WithAlias("e")
+                    .WithDescription("Shows examples of status module.")
```
Comment (EluciusFTW) [inline src/spectrecoff-cli/Program.fs:269]: "... of the ..."  ?

File: src/spectrecoff-cli/commands/Status.fs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        let thinkingProcess (context: StatusContext) =
+            task {
+                do! Task.Delay(5000)
+                context.Status <- "More thinking"
+                context.SpinnerStyle <- Style.Parse("green")
+                context.Spinner <- Spinner.Known.Balloon2
+                do! Task.Delay(5000)
+            }
+
+        (startWithSpinnerStyle "Thinking" Spinner.Known.Pong (Style.Parse("red")) thinkingProcess).Wait()
```
Comment (EluciusFTW) [inline src/spectrecoff-cli/commands/Status.fs:25]: Is the type expected here the same as the one returned by the `toSpectreStyle `function in `Styling.fs`? If so, we don't expose the style anywhere else, so I'd suggest acception a `look` here as well. And I'd call it `startWithCustomSpinner` instead.

**Review by EluciusFTW — COMMENTED** [review]
Nice! Only some smaller things I noted.

File: src/spectrecoff-cli/Program.fs (hunk 1 above)
Comment (DanielmTheDev) [reply src/spectrecoff-cli/Program.fs]: Done

File: src/spectrecoff-cli/commands/Status.fs (hunk 2 above)
Comment (DanielmTheDev) [reply src/spectrecoff-cli/commands/Status.fs]: That is a good point, changed as requested. Thanks @EluciusFTW !

File: src/spectrecoff-cli/commands/Status.fs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+type StatusExample() =
+    inherit Command<StatusSettings>()
+    interface ICommandLimiter<StatusSettings>
+
+    override _.Execute(_context, _settings) =
+        let thinkingProcess (context: StatusContext) =
+            task {
+                do! Task.Delay(5000)
+                context.Status <- "More thinking"
```
Comment (EluciusFTW) [inline src/spectrecoff-cli/commands/Status.fs:19]: Maybe we should offer a functions:

```fsharp
updateContext newMessage newSpinner newLook context = 
    context.Status <- newMessage
    context.SpinnerStyle <- newLook |> toSpectreStyle
    context.Spinner <- newSpinner
    context
```

and then go one step further:

```fsharp
type sc = 
  { Message: string
    Spinner: string option
    Look: Look option }
```

which you pass as argument into the start/update method (i.e., have one method which only accepts string like start now, one which takes an instance of this type), and then, on update, if you only want to change the message, you can do 

`context |> update { sc with Message = "Step 2" }`

(Please replace sc with appropriate type name and variable name resp.)

Comment (DanielmTheDev) [reply src/spectrecoff-cli/commands/Status.fs:19]: This is a great idea :) I will implement it once my pesky real work releases me for a moment :)

Comment (DanielmTheDev) [reply src/spectrecoff-cli/commands/Status.fs]: I changed according to your suggestion. Somewhat unsure about the formatting, but I will not tell you where :) Until then, patiently awaiting your feedback @EluciusFTW

File: src/spectrecoff/Status.fs [hunk 4]
```diff
@@ -4,20 +4,46 @@ module SpectreCoff.Status
...
+        match customerSpinner.Spinner with
+        | Some spinner -> spinner
+        | None _ -> context.Spinner
+
+    context.SpinnerStyle <-
+        match customerSpinner.Look with
+        | Some look -> look |> toSpectreStyle
+        | None _ -> context.SpinnerStyle
+
+    context.Spinner
```
Comment (EluciusFTW) [inline src/spectrecoff/Status.fs:49]: Why are you returning the spinner, not the context?

File: src/spectrecoff/Status.fs [hunk 5]
```diff
@@ -4,20 +4,46 @@ module SpectreCoff.Status
...
+type CustomSpinner =
+    { Message: string
+      Spinner: Spinner Option
+      Look: Look Option }
+
 let start statusText (operation: StatusContext -> Task<unit>) =
     task { return! AnsiConsole.Status().StartAsync(statusText, operation) }
 
-let startWithSpinner statusText spinner (operation: StatusContext -> Task<unit>) =
+let startWithCustomSpinner customerSpinner (operation: StatusContext -> Task<unit>) =
```
Comment (EluciusFTW) [inline src/spectrecoff/Status.fs:15]: Just `spinner `is fine, imo

**Review by EluciusFTW — APPROVED** [review]
Just two mor tiny things, approved.

File: src/spectrecoff/Status.fs (hunk 4 above)
Comment (DanielmTheDev) [reply src/spectrecoff/Status.fs]: Makes no sense, changed

File: src/spectrecoff/Status.fs (hunk 5 above)
Comment (DanielmTheDev) [reply src/spectrecoff/Status.fs]: Renamed (hungarian notation nono, as you yourself taught me once).

**Pull request merged**